django/db/models/constraints.py

```diff
@@ -69,14 +73,21 @@ def deconstruct(self):
         return path, args, kwargs
 
 
+DEFERRABLE_INITIALLY_DEFERRED = 'deferred'
+DEFERRABLE_INITIALLY_IMMEDIATE = 'immediate'
```

Comment:
Any thoughts about using a `Deferrable(Enum)` for this purpose?

```python
class Deferrable(Enum):
    DEFERRED = 'deferred'
    IMMEDIATE  = 'immediate'
```

The same enum could be used to replace the `DEFERRED` and `IMMEDIATE` below and to enforce a valid value is passed to `UniqueConstraint.__init__`.

Reply:
I considered it briefly, but didn't come to any particular decision. I like the idea of using it for both `UniqueConstraint` and `set_constraints` though.